Sources/Ink/Internal/List.swift

```diff
@@ -16,7 +19,22 @@ internal struct List: Fragment {
 
     private static func read(using reader: inout Reader,
                              indentationLength: Int) throws -> List {
-        var list = List(isOrdered: reader.currentCharacter.isNumber)
+        let isOrdered = reader.currentCharacter.isNumber
+
+        var list: List
+        if isOrdered {
+            let startIndex = reader.currentIndex
+            defer { reader.moveToIndex(startIndex) }
+            
+            let startingIndexString = try reader.readCharacters(matching: \.isNumber, max: 9)
+            let startingIndex = Int(startingIndexString) ?? 1
```

Comment:
It's quite confusing to have one variable called `startIndex` and another one called `startingIndex`, but I think by getting rid of the above `defer` statement that'll be solved 🙂

Reply:
I renamed it to `firstNumber`, does that clear up the confusion?